XTS/WS-C/dev/src/com/arjuna/webservices11/wscoor/client/WSCOORClient.java

```diff
@@ -41,12 +42,27 @@
     private static synchronized ActivationService getActivationService()
     {
         if (activationService.get() == null) {
-            // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
-            activationService.set(new ActivationService());
+            setActivationService();
         }
         return activationService.get();
     }
 
+    private static synchronized void setActivationService()
+    {
+        // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
+        if (System.getSecurityManager() == null) {
```

Comment:
How often is  setActivationService called - if lots of times then there might be some performance benefit from cacheing. If not then no point. This is a general point about sprinkling privileged actions throughout our code base and it might be beneficial to make sure we are doing things efficiently.